osu.Game.Rulesets.Osu/Edit/Blueprints/Sliders/SliderPlacementBlueprint.cs

```diff
@@ -171,7 +171,7 @@ protected override bool OnDragStart(DragStartEvent e)
                 return base.OnDragStart(e);
 
             // Only enter drawing mode if no additional control points have been placed.
-            if (HitObject.Path.ControlPoints.Count > 2)
+            if (HitObject.Path.ControlPoints.Count > 2 || HitObject.Path.ControlPoints.Last() != cursor)
```

Comment:
After testing with this change, I cannot start free-hand drawing at all locally anymore - presumably because the cursor cp gets placed before OnDragStart gets hit, causing this check to always be true on my end.

Reply:
Yeah hold your horses. I'm adding tests and fixing things.

Maybe come back in an hour or so.

Reply:
Should be in a good state now, at least for that particular issue. There's one more minor discrepancy I've found with the curve type choice that I'll investigate further.  